sklearn/inspection/_ice.py

```diff
@@ -0,0 +1,491 @@
+    >>> gb = GradientBoostingClassifier(random_state=0).fit(X, y)
+    >>> individual_conditional_expectation(gb, features=[0], X=X,
+    ...       percentiles=(0, 1), grid_resolution=2) # doctest: +SKIP
+    (array([[[2.19017313e-05, 5.16319395e-01],
+        [4.83680605e-01, 9.99978098e-01]]]), [array([0., 1.])])
+    """
+    X, response_method, _ = _validate_pdp_ice_parameters(estimator, X,
+                                                         features,
+                                                         response_method,
+                                                         'brute')
+
+    features_indices = np.asarray(
+        _get_column_indices(X, features), dtype=np.int32, order='C'
+    ).ravel()
+
+    grid, values = _grid_from_X(
+        _safe_indexing(X, features_indices, axis=1), percentiles,
+        grid_resolution
+    )
+
+    predictions = _ice_brute(
+        estimator, grid, features_indices, X, response_method
+    )
+
+    # reshape predictions to
+    # (n_outputs, n_instances, n_values_feature_0, n_values_feature_1, ...)
+    predictions = predictions.reshape(
+        -1, X.shape[0], *[val.shape[0] for val in values])
+
+    if centre:
+        for i, instances in enumerate(predictions):
+            for j, instance in enumerate(instances):
+                first_element = instance.item(0)
+                predictions[i][j] = instance - first_element
+
+    return predictions, values
+
+
+def plot_individual_conditional_expectation(estimator, X, features,
+                                            feature_names=None, target=None,
+                                            response_method='auto', n_cols=3,
+                                            grid_resolution=100, centre=True,
+                                            percentiles=(0.05, 0.95),
+                                            n_jobs=None, verbose=0,
+                                            line_kw=None, ax=None):
+    """Individual Conditional Expectation (ICE) plots.
+
+    The ``len(features)`` plots are arranged in a grid with ``n_cols``
+    columns. The deciles of the feature values will be shown with tick marks
+    on the x-axes.
+
+    .. note::
+
+        :func:`plot_individual_conditional_expectation` does not support using
+        the same axes with multiple calls. To plot the ICE curves for multiple
+        estimators, please pass the axes created by the first call to the
+        second call::
+
+          >>> from sklearn.inspection import (
+          ...                   plot_individual_conditional_expectation)
+          >>> from sklearn.datasets import make_friedman1
+          >>> from sklearn.linear_model import LinearRegression
+          >>> X, y = make_friedman1()
+          >>> est = LinearRegression().fit(X, y)
+          >>> disp1 = plot_individual_conditional_expectation(est,
+          ...                                 X)  # doctest: +SKIP
```

Comment:
indentation here is awkward. Rather, put a line break after (

Reply:
Done in https://github.com/scikit-learn/scikit-learn/pull/16164/commits/a3d6a634245918bf28f75758fdadc6df893ea197